dspace-server-webapp/src/test/java/org/dspace/app/rest/ResourcePolicyRestRepositoryIT.java

```diff
@@ -2202,6 +2202,518 @@ public void patchAddNameTest() throws Exception {
+        ReplaceOperation replaceOperation = new ReplaceOperation("/policyType", newPolicyType);
+        ops.add(replaceOperation);
+        String patchBody = getPatchContent(ops);
+
+        String authToken = getAuthToken(eperson1.getEmail(), "qwerty01");
+        getClient(authToken).perform(patch("/api/authz/resourcepolicies/" + resourcePolicy.getID())
+            .content(patchBody)
+            .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$", Matchers.allOf(
+                hasJsonPath("$.policyType", is(newPolicyType)),
+                hasJsonPath("$.action", is(Constants.actionText[resourcePolicy.getAction()])))));
+
+        getClient(authToken).perform(get("/api/authz/resourcepolicies/" + resourcePolicy.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$", Matchers.allOf(
+                hasJsonPath("$.action", is(Constants.actionText[resourcePolicy.getAction()])),
+                hasJsonPath("$.policyType", is(newPolicyType)))));
+    }
+
+    @Test
+    public void patchReplacePolicyTypeUnauthenticatedTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        Community community = CommunityBuilder.createCommunity(context).build();
+
+        Collection collection = CollectionBuilder.createCollection(context, community).build();
+
+        Item item = ItemBuilder.createItem(context, collection).build();
+
+        ResourcePolicy resourcePolicy = ResourcePolicyBuilder.createResourcePolicy(context)
+            .withAction(Constants.WRITE)
+            .withDspaceObject(item)
+            .withPolicyType(ResourcePolicy.TYPE_CUSTOM)
+            .build();
+
+        context.restoreAuthSystemState();
+
+        String newPolicyType = ResourcePolicy.TYPE_SUBMISSION;
+
+        List<Operation> ops = new ArrayList<Operation>();
+        ReplaceOperation replaceOperation = new ReplaceOperation("/policyType", newPolicyType);
+        ops.add(replaceOperation);
+        String patchBody = getPatchContent(ops);
+
+        getClient().perform(patch("/api/authz/resourcepolicies/" + resourcePolicy.getID())
+            .content(patchBody)
+            .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
+            .andExpect(status().isUnauthorized());
+
+        String authToken = getAuthToken(admin.getEmail(), password);
+        getClient(authToken).perform(get("/api/authz/resourcepolicies/" + resourcePolicy.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$", Matchers.allOf(
+                hasJsonPath("$.action", is(Constants.actionText[resourcePolicy.getAction()])),
+                hasJsonPath("$.policyType", is(resourcePolicy.getRpType())))));
+    }
+
+    @Test
+    public void patchReplacePolicyTypeForbiddenTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        EPerson eperson1 = EPersonBuilder.createEPerson(context)
```

Comment:
Here already defined user in field "eperson" of base class can be used without creating a new one